demonstrations/tutorial_zx_calculus.py

```diff
@@ -0,0 +1,799 @@
+Choi-Jamiolkowski equivalent to the identity linear map.
+
+Let's simplify the diagram by applying some rewriting rules. The first step is to fuse the :math:`a` state with the
+X-spider of the CNOT. We also merge the Hadamard gate with the :math:`b` state, because together it represents a
+Z-spider. Then we can fuse the three Z-spiders by simply adding their phases. After that we see that the Z-spider
+phase vanishes modulo of :math:`2\pi` and can therefore be simplified using the identity rules. Then we can fuse the
+two X-spiders by adding their phase. We notice that the phase again vanishes modulo :math:`2\pi` and we can get rid
+of the last X-spider. Teleportation is a simple wire connecting Alice and Bob!
+
+.. figure:: ../demonstrations/zx_calculus/teleportation.jpg
+    :align: center
+    :width: 70%
+
+    The teleportation ZX-diagram simplified.
+
+
+ZXH-calculus
+------------
+
+The universality of the ZX-calculus does not guarantee the existence of a simple representation, even for simple
+linear maps. For example, the Toffoli gate (the quantum AND gate) requires around 25 spiders (Z and X)! We previously
+introduced the Hadamard gate as a yellow box, this motivates the introduction of a new generator: the multi-leg H-box,
+defined as follows:
+
+.. figure:: ../demonstrations/zx_calculus/h_box.jpg
+    :align: center
+    :width: 70%
+
+    The H-box, a third generator.
+
+The parameter :math:`a` can be any complex number, and the sum  is over all :math:`i1, ... , im, j1, ... , jn \in {0,
+1}`, therefore an H-box represents a matrix where all entries are equal to :math:`1`, except for the bottom right
+element, which is \ :math:`a`. This will allow us to concisely express the Toffoli gate, as we will see shortly.
+
+An H-box with one input wire and one output wire, with :math:`a=-1`, is an Hadamard gate up to global phase. Thus,
+we omit the parameter when it is equal to :math:`-1`. The Hadamard gate is sometimes represented by a blue edge
+rather than a box.
+
+Thanks to the introduction of the multi-leg H-box, the Toffoli gate can be represented with three Z spiders and three
+H-boxes — two simple Hadamard gates and one three-ary H-box — as shown below:
+
+.. figure:: ../demonstrations/zx_calculus/toffoli.jpg
+    :align: center
+    :width: 70%
+
+    Toffoli
+
+The addition of the multi-leg H-box together with an additional set of rewriting rules form the ZXH calculus. You can
+find more details and the rewriting rules in the literature [#East2021]_.
+
+Let's show that this ZXH-diagram is indeed a Toffoli-gate. This operation is defined by conditionally applying a
+X-gate on the target wire, it means that only the state 110 and 111 will not map to themselves (110 to 111 and 111 to
+110). We will show that if one provides the state 11 on the two first wires, it results by a bit flip on the third wire
+(X-gate). For that purpose, we need to add a new rewriting rule that is part of the ZXH-calculus, the absorb rule:
```

Comment:
```suggestion
(X-gate). For that purpose, we need to add a new rewriting rule that is part of the ZXH-calculus: the absorb rule.
```

Probably a good idea to bold / italicize all rewriting rule names.